tests/console/pam.pm

```diff
@@ -0,0 +1,61 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: This PM is to run a suite of tests about PAM, and gets a TAP
+#          format result. Four files(pam.sh, pam_test, pam_test.sh,
+#          system-default) are included. pam.sh is a "bats" script, and
+#          includes 25 tests for PAM. The other files are from the project
+#          "pam_test" which can be used to test a PAM stack for authentication
+#          and password change. The link: https://github.com/pbrezina/pam-test
+#   Steps:
+#       - add qa-head repo and install the tool "bats" from qa-head repo.
+#         Bats is a TAP-compliant testing framework for Bash.
+#       - download "pam_test" dir(automated tests of PAM) from "data" dir
+#         on openqa host if "prepare_test_data" perl module desn't run.
+#       - run the suite of tests in SUT.
+# Maintainer: Jun Wang <jgwang@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils 'is_sle';
+
+sub run {
+    select_console 'root-console';
+
+    my $version      = get_required_var('VERSION');
+    my $qa_head_repo = is_sle() ? "http://download.suse.de/ibs/QA:/Head/SLE-$version/" : 'https://download.opensuse.org/tumbleweed/repo/oss/';
```

Comment:
```suggestion
    my $qa_head_repo = is_sle() ? "http://download.suse.de/ibs/QA:/Head/SLE-$version/" : "https://download.opensuse.org/$version/repo/oss/";
```

Reply:
I think the tumbleweed OSS repo is enabled by default.

Reply:
I am not sure what that line is supposed to achieve for !sle. Adding this repo is probably a bad idea as we want to test products before release. By this on Tumbleweed you are mixing in content from the old, last published repo. On leap you are adding an incompatible repo

Reply:
@okurz I think you are right. You know, the test pam.pm is depended on the package pam-test, but there is no this package in openSUSE repos, so I asked  Martin Plusky to help me build the package and put it into the above repo openSUSE:Factory. Right now, I need to build this package for openSUSE if I want to support openSUSE, and put this package into a correct repo, right? or any other solution? please give me some suggestions.

Reply:
Maybe a subproject of https://build.opensuse.org/project/show/devel:openSUSE:QA is the right target

Reply:
> Maybe a subproject of https://build.opensuse.org/project/show/devel:openSUSE:QA is the right target

@okurz  Thank you :smile: . I notice that you are a maitainer of the project devel:openSUSE:QA, and right now the package pam-test is in the project openSUSE:Factory, how do I add this package into the subproject of devel:openSUSE:QA? I have built this package pam-test in my home project https://build.opensuse.org/package/show/home:jgwang/pam-test, but I don't know how to add it into other projects. Could you help me?

Reply:
@okurz @foursixnine  I have updated my code. Once devel:openSUSE:QA is ready, this test can work for openSUSE, but right now I don't schedule this test for openSUSE.

Reply:
now I am confused. There is https://build.opensuse.org/package/show/openSUSE:Factory/pam-test , why not use that?

Reply:
@okurz you mean that I don't need to create a project and build pam-test? I know that pam-test is in openSUSE:Factory, but I don't know how to link it to devel:openSUSE:QA, I know you are one of the maintainers of this project, could you help me link pam-test to devel:openSUSE:QA?

Reply:
> Once devel:openSUSE:QA is ready, this test can work for openSUSE, but right now I don't schedule this test for openSUSE.

Then let's remove the part to handle openSUSE, and move forward with this. Sorry for jumping back and forth on this topic.

Reply:
> Then let's remove the part to handle openSUSE, and move forward with this. Sorry for jumping back and forth on this topic.

It's OK. I will remove the lines, and add some comments to explain that opensuse is not supported right now.

Reply:
when the package is already in openSUSE:Factory then use this package. This just means you do not need to add any additional repo. It should be supported still though